src/main/kotlin/org/rust/ide/intentions/JoinWildcardsIntention.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.endOffset
+import org.rust.lang.core.psi.ext.startOffset
+
+class JoinWildcardsIntention : RsElementBaseIntentionAction<List<RsPatWild>>() {
+    override fun getFamilyName(): String = "Join wildcards into `..`"
+    override fun getText(): String = familyName
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): List<RsPatWild>? {
+        val patList = when (val pat = element.ancestorStrict<RsPat>() ?: return null) {
+            is RsPatTup -> pat.patList
+            is RsPatTupleStruct -> pat.patList
+            is RsPatSlice -> pat.patList
+            // note that RsOrPat also has (and allows) multiple _ wildcards.
+            // it does, however, not support `..`, so we leave it out for this inspection.
+            else -> return null
+        }
+        val wildcards = patList.takeLastWhile { it is RsPatWild }.map { it as RsPatWild }
```

Comment:
Some remarks:
1) It does not have to be only at the end, i.e. `let (_, _, a) = (1, 2, 3);`. Basically any sequence of 1 or more wild patterns can be replaced with a single `..`.
2) You should also check if `..` is already not present in the pat.
3) The intention should be applicable on a wild pat, not on its whole parent pat, because of 1). Consider e.g. `(_, _, a, _, _) = (1, 2, 3, 4, 5)`. You can only change the first two or the last two underscores to `..`, but not both.